### ISSUE TITLE
Logs streams, worker events, REPL...

### ISSUE THREAD

**Issue by seebees** [isaacs/cluster-master#22]
This project has not had a lot of activity.  I would like to see some new features like:

* Transform streams for workers stdout: So you could append process id etc to logs bubbling up from the workers
* onMessage is supported, but what about `exit`, `online` etc.
* `.exit` in the REPL is not exactly right, I think that sock.end() is what people are looking for, update documentation/implementation.
* Additional REPL features (some listed elsewhere), but I would like to be able to attach stdout for a given worker to the REPL...

Should I make some PR's or should I fork the project and start anew?

**Comment by isaacs** [comment]
I don't use this any more.  Anyone who lands good PRs, I'll add as collaborators.  Go nuts :)

**Comment by seebees** [comment]
@isaacs, hum... Why don't you use it anymore?  Do you use something else or do you just use the built in cluster module?

It seems to me that this direction offers some things for production grade monitoring.  I'll fork this look at any suggestions I can (hopefully ;) get out of you and set up some PR's...

**Comment by isaacs** [comment]
> Why don't you use it anymore? Do you use something else or do you just use the built in cluster module?

It's just easier spin up a bunch of independent VMs, and put a load balancer in front of them, especially since you usually want a CDN in front of any production site anyway.  Then clusters just kind of make things confusing.

**Comment by seebees** [comment]
Yup... that sounds right.  The cluster stuff seems more like an ops function since it provides faster recovery, fault tolerance, maybe some monitoring magic...  I'll start playing around and we shall see.

Ideally, it should be that a dev-ops person could drop in something like cluster-master on top of an arbitrary application and it should just work...  That's my goal anyway.
